### DISCUSSION TITLE
Recommended way to test Carbon in `expect`

### DISCUSSION THREAD

**Discussion by JasonTheAdams** [pestphp/pest#326] [Q&A] [answered]
Hi! 👋

I was recently trying to do something like this:
```php
// Not same instance
expect($token)
    ->expiration->toBe($expirationDate);

// Doesn't seem to know how to handle this
expect($token)
    ->expiration->format('U')->toBe($expirationDate->format->('U'));

// Works
assertSame($expirationDate->format('U'), $token->expiration->format('U'));
```

I like the `expect()` syntax, but there doesn't seem to be a suggested way of working with dates (and Carbon, specifically) in Pest. How are others handling this?

Thanks!

**Comment by nunomaduro** [comment, marked as answer]
cc @lukeraymonddowning

**Reply by lukeraymonddowning** [reply]
Isn't this nested HOE? As far as I can tell, this will "just work" in the next release 🎉

**Reply by JasonTheAdams** [reply]
Yeah, I saw that PR and wondered if this fell into that. I'd certainly be happy to test it once the release comes out!

**Reply by lukeraymonddowning** [reply]
It has been released with 1.7

**Reply by JasonTheAdams** [reply]
Works with 1.7! 🎉

**Comment by lukeraymonddowning** [comment]
Any reason you can't use toEqual in the first example?

**Reply by nunomaduro** [reply]
Fell free to release already if you want.

**Reply by JasonTheAdams** [reply]
> Any reason you can't use toEqual in the first example?

Equals with Carbon instances can be finicky. I've found the unix timestamp to be a more reliable comparison. 🤷‍♂️

**Comment by clemblanco** [comment]
Just to clarify, we still have to use something like:

```php
expect($someDateTime->unix())->toBe($expectedDateTime->unix());
```

and 

```php
expect($someDateTime)->toBe($theExpectedDateTime);
```

will never work, right?

**Reply by fabio-ivona** [reply]
Not natively

Just use this https://github.com/defstudio/pest-plugin-laravel-expectations
